### PR TITLE
Onboarding

### LINKED ISSUE #371: Onboarding Slides for Apollos
## Acceptance Criteria

- [ ] It should [SPEC_DETAIL]

## Context

Given [CONTEXT] when [ACTION] then [RESULT]

### PR DESCRIPTION
**Pull request by richarddubay**
Fixes #371 

## Creator

- [x] Build relevant tests if any apply
- [x] Ensure there are no new warnings
- [ ] Upload an animated GIF showcasing the solution (if it applies)
- [x] Set a relevant reviewer

## Reviewer

- [ ] Run `test` and make sure all tests pass
- [ ] Review function and form on iOS
- [ ] Review function and form on Android
- [ ] Review function and form on Web
- [ ] Review new test logic

### CODE REVIEW

**Comment by richarddubay** [comment]
@conrad-vanl and/or @burkeshartsis - two questions:

1. How should I handle the web version of this? Better...what should I do in the `index.web.js` file since this isn't showing on the web?
2. Ideas on how to get this working? A modal/route/whatever. Do you guys want to handle that?

I'm technically still waiting on the slides. I'm meeting tomorrow to show them what I've done and what I need them to do. But that shouldn't stop us from getting this thing moving. 

cc: @IsaacHardy

**Comment by conrad-vanl** [comment]
> How should I handle the web version of this? Better...what should I do in the index.web.js file since this isn't showing on the web?

The web version can basically just be a function that returns null or children, kinda like other components we don't support on web: https://github.com/NewSpring/Apollos/blob/alpha/src/%40ui/Audio/index.web.js

> Ideas on how to get this working? A modal/route/whatever.

This is just spit-balling, but what I'm thinking we could do is have a component that wraps what you've started in a React-Native [`Modal`](https://facebook.github.io/react-native/docs/modal.html). We can store a value in [`AsyncStorage`](https://facebook.github.io/react-native/docs/asyncstorage.html) on whether this modal is visible or not. When the onboarding is dismissed, we set that value in asyncstorage so that on subsequent loads the onboarding modal isn't visible.

This component can be rendered outside of the `AppRouter`, maybe one of the components in this list: https://github.com/NewSpring/Apollos/blob/alpha/src/index.js#L30. While it's waiting for `AsyncStorage` to get the value, it can just render [`AppLoading`](https://docs.expo.io/versions/latest/sdk/app-loading.html), which will keep the splash screen visible.

**Comment by richarddubay** [comment]
@conrad-vanl or @burkeshartsis If you guys want to take over at this point, I'd be okay with that. I've made an `OnboardingModal` component to display the regular `Onboarding` component. The `Onboarding` component has the new artwork. The only thing it's missing is to fill in the `Done` function so that something happens when you tap the `Done` button on the last slide.

If you guys don't have the time to look into this, that's fine and I can continue to. Just let me know. I figured you guys would be able to get this up and running faster than me and possibly get it into the build before we ship next Monday.

I appreciate all you do! You're awesome!

**Comment by richarddubay** [comment]
@conrad-vanl No worries. I'll just keep pecking away at it until you do. You may end up ripping out everything, but that would be a good learning experience for me! ;)

**Comment by conrad-vanl** [comment]
hey @richarddubay I went ahead and hooked this up. I decided to try using `apollo-link-state` / our client-side graphql store to do so - that way we didn't have this isolated `AsyncStorage` inside of one component for state, _and_ it allowed me to put a "Show Onboarding" button in the settings menu so that we can always re-open it

**Comment by conrad-vanl** [comment]
I also tweaked the image styles a bit for iPhone X. This isn't perfect - it looks like this slider library you're using is designed so that the images used don't have artwork that go all the way to the edge, and so I might suggest having art not have artwork clipped on any of the images, as I end up with stuff like this on really tall screens:

image

If i stretched the artwork all the way top to bottom, which I'm not sure I can even do with this library, then the other images would be awkwardly cropped. FWIW, I did this because when I first jumped into this branch, the text on slides 2-7 were _really_ close to the notch and didn't look very good to me.

**Comment by richarddubay** [comment]
@conrad-vanl I noticed when I started trying to hook this up that what I was seeing in the actual modal differed from what I was seeing in Storybook, so I figured they would need some tweaking anyway. At least the `Onboarding` component would. I can get the guy who made these look at editing them so it gets the images away from the edges.

**Comment by richarddubay** [comment]
@conrad-vanl The first and last slides have been updated to not have any art on the edges. This is looking pretty good I think. I'd love to get this moving. cc: @IsaacHardy

**Comment by richarddubay** [comment]
It even looks pretty good on an iPhone 5S!

screenshot 2018-04-05 10 23 42
screenshot 2018-04-05 10 23 46
screenshot 2018-04-05 10 23 54

**Comment by IsaacHardy** [comment]
Everything looks good. Before we merge, I'd like to discuss my comment above.

File: src/@ui/Onboarding/index.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  },
+  {
+    key: 'eight',
+    image: require('../../assets/onboarding/8-Get-Started.jpg'),
+    imageStyle: styles.image,
+    backgroundColor: '#6bac43',
+  },
+];
+
+const renderDoneButton = () => <Chip pill title="Done" type="secondary" />;
```
[the remark this answers is not in the document]
Comment (conrad-vanl) [reply src/@ui/Onboarding/index.js]: I like "Go"! Nice and short!

**Comment by richarddubay** [comment]
screenshot 2018-04-05 14 35 43

**Comment by conrad-vanl** [comment]
looks good @richarddubay ! just waiting for that commit to be pushed up and we can mark this `merge`!

**Pull request merged**